Ascend-Server.api/Controllers/LoginController.cs

```diff
@@ -0,0 +1,50 @@
+using Models; 
+using Services; 
+using Exceptions; 
+using Microsoft.AspNetCore.Mvc;
+
+[Route("api/[controller]")]
+[ApiController]
+public class LoginController : ControllerBase
+{
+    private readonly IUserService _userService; 
+
+    public LoginController(IUserService userService)
+    {
+        _userService = userService; 
+    }
+
+    [HttpPost]
+    public IActionResult Post(LoginRequest request)
+    {
+        
+    User? user; 
+    try
+    {
+     user = _userService.FindUserByEmail(request.Email!);
+
+     _userService.VerifyPassword(user!, request.Password!); 
+
+    }
+    catch(EmailDoesNotExistException e)
+    {
+     return NotFound(e.Message); 
+    }
+    catch(InvalidPasswordException e)
+    {
+        Unauthorized(e.Message); 
+    }
+    catch(Exception)
+    {
+        return new BadRequestResult(); 
+    }
```

Comment:
You'll want to be careful with this type of flow. Two things:
1. providing a different error message or workflow when the email address isn't valid vs a password isn't valid allows an attacker to gain additional information and gives them a way to focus on one area. You always want the same message on if a username or a password is not valid.
2. This is using exceptions for process flow. Some would consider other places where you return things like UserNotFoundException to also be process flow with exceptions but that is a more common practice and something we don't need to talk about alternatives for today. The concern is around calling a verify password method and returning exceptions for any negative cases. We don't really need to do that and exceptions are quite expensive in terms of performance. Instead, you could use a result pattern where the verify password passes back a state object indicating success or failure with perhaps a bit more information if you need it. In this case you may not even need anything more than returning back a boolean if you keep your messages generic.